### PR TITLE
Ignore signature help trigger characters in comments and strings

### LINKED ISSUE #3385: Trigger keys for signature help are not ignored within comments
Was looking to matching C#'s LS behavior for TypeScript. I noticed that signature help trigger keys are ignored within strings but not within comments.

triggerkeys-csharp

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Fixes #3385.

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
Hey @rchande and @brettfo, think you can take a quick look at this?

File: src/EditorFeatures/VisualBasicTest/SignatureHelp/AbstractVisualBasicSignatureHelpProviderTests.vb [hunk 1]
```diff
@@ -11,11 +11,11 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.SignatureHelp
 
         ' We want to skip script testing in all VB stuff for now.
 
-        Protected Overrides Sub Test(markupWithPositionAndOptSpan As String, Optional expectedOrderedItemsOrNull As IEnumerable(Of SignatureHelpTestItem) = Nothing, Optional usePreviousCharAsTrigger As Boolean = False, Optional sourceCodeKind As Microsoft.CodeAnalysis.SourceCodeKind? = Nothing, Optional experimental As Boolean = False)
+        Protected Overrides Sub Test(markupWithPositionAndOptSpan As String, Optional expectedOrderedItemsOrNull As IEnumerable(Of SignatureHelpTestItem) = Nothing, Optional usePreviousCharAsTrigger As Boolean = False, Optional sourceCodeKind As Microsoft.CodeAnalysis.SourceCodeKind? = Nothing, Optional experimental As Boolean = False, Optional expectSuccess As Boolean = True)
             If (sourceCodeKind.HasValue) Then
-                MyBase.Test(markupWithPositionAndOptSpan, expectedOrderedItemsOrNull, usePreviousCharAsTrigger, sourceCodeKind, experimental)
+                MyBase.Test(markupWithPositionAndOptSpan, expectedOrderedItemsOrNull, usePreviousCharAsTrigger, sourceCodeKind, experimental, expectSuccess)
```
Comment (rchande) [inline src/EditorFeatures/VisualBasicTest/SignatureHelp/AbstractVisualBasicSignatureHelpProviderTests.vb]: Passing in an empty IEnumerable<SignatureHelpItem> should assert that no items were produced. I don't think you need to add the ```expectSuccess``` parameter.

**Comment by DanielRosenwasser** [comment]
Hey @rchande, I addressed that comment; think we're good to go?

**Comment by rchande** [comment]
@DanielRosenwasser :+1: Thanks!

**Comment by jaredpar** [comment]
CC @jasonmalinowski, @dpoeschl, @Pilchie, @basoundr 

Can one of you all add the second sign off?

**Comment by jasonmalinowski** [comment]
:+1:. Has somebody given this an internal run and/or buddy-test?

**Comment by brettfo** [comment]
I apologize for letting this slip my radar for so long.  I'm doing a buddy-test now and running our full suite of internal tests, too, and should be able to respond shortly.

**Comment by brettfo** [comment]
All internal tests pass.  I'll kick off one more Jenkins run as a sanity check then I'll merge it.

**Comment by brettfo** [comment]
retest this please

**Pull request merged**